src/Shortcodes/FileShortcodeProvider.php

```diff
@@ -102,6 +102,8 @@ public static function handle_shortcode($arguments, $content, $parser, $shortcod
             $url = $record->Link();
         }
 
+        $url = $url ?? '';
+
```

Comment:
I've tested without this and it doesn't seem to change anything... seems to not be needed?

Reply:
`File::getURL()` returns null https://github.com/silverstripe/silverstripe-assets/blob/7605cda151919e8638c3b300e4864d6c3748c349/src/File.php#L896.
And then we set this value in `$markup` in cache. After we return $item['markup'] in `getCachedMarkup` method.
So probably, there are some cases where we can get null from `getCachedMarkup`
I'll add checking in the `getCachedMarkup` method.

Reply:
Updated